### ISSUE TITLE
When cancelling websocket, exception thrown: aiohttp._websocket.reader_c.WebSocketDataQueue' object has no attribute 'is_eof'

### ISSUE THREAD

**Issue by egoldschmidt** [fafhrd91/aiohttp#9880]
### Describe the bug

When closing a websocket connection (in response to task.cancel() I think), I am seeing errors of this form:

```
AttributeError: 'aiohttp._websocket.reader_c.WebSocketDataQueue' object has no attribute 'is_eof'
^^^^^^^^^^^^^^^^^^^^
or (self._payload is not None and not self._payload.is_eof())
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/client_proto.py", line 55, in should_close
^^^^^^^^^^^^^^^^^^^^^
if self._force_close or should_close or protocol.should_close:
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/connector.py", line 705, in _release
self._connector._release(self._key, self._protocol)
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/connector.py", line 197, in release
self._connection.release()
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 945, in __del__
Traceback (most recent call last):
Exception ignored in: <function ClientResponse.__del__ at 0x7faf9467ff60>
```

A cursory reading of [_websocket/reader.py](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/_websocket/reader_py.py#L52) seems to suggest that indeed there is no `is_eof` function.

### To Reproduce

Start a websocket and call .cancel() on the task or stop the event loop

### Expected behavior

No error

### Logs/tracebacks

```python-traceback
AttributeError: 'aiohttp._websocket.reader_c.WebSocketDataQueue' object has no attribute 'is_eof'
^^^^^^^^^^^^^^^^^^^^
or (self._payload is not None and not self._payload.is_eof())
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/client_proto.py", line 55, in should_close
^^^^^^^^^^^^^^^^^^^^^
if self._force_close or should_close or protocol.should_close:
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/connector.py", line 705, in _release
self._connector._release(self._key, self._protocol)
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/connector.py", line 197, in release
self._connection.release()
File "/home/appuser/.venv/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 945, in __del__
Traceback (most recent call last):
Exception ignored in: <function ClientResponse.__del__ at 0x7faf9467ff60>
```
```


### Python Version

```console
$ python --version

Python 3.12.7
```

### aiohttp Version

```console
$ python -m pip show aiohttp

Name: aiohttp
Version: 3.11.0
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author:
Author-email:
License: Apache 2
Location: /Users/evan/Projects/portola/livekit-worker/.venv/lib/python3.12/site-packages
Requires: aiohappyeyeballs, aiosignal, attrs, frozenlist, multidict, propcache, yarl
Required-by: livekit-agents, livekit-api
```

### multidict Version

```console
$ python -m pip show multidict

Name: multidict
Version: 6.1.0
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /Users/evan/Projects/portola/livekit-worker/.venv/lib/python3.12/site-packages
Requires:
Required-by: aiohttp, yarl
```

### propcache Version

```console
$ python -m pip show propcache

Name: propcache
Version: 0.2.0
Summary: Accelerated property cache
Home-page: https://github.com/aio-libs/propcache
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /Users/evan/Projects/portola/livekit-worker/.venv/lib/python3.12/site-packages
Requires: 
Required-by: aiohttp, yarl
```

### yarl Version

```console
$ python -m pip show yarl

Name: yarl
Version: 1.17.1
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: /Users/evan/Projects/portola/livekit-worker/.venv/lib/python3.12/site-packages
Requires: idna, multidict, propcache
Required-by: aiohttp
```

### OS

Debian

### Related component

Client

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by egoldschmidt** [comment]
Oh one important note: this was definitely not happening on 3.10.10

**Comment by bdraco** [comment]
Looks like we don’t have a test for this case.  Should be an easy fix

**Comment by bdraco** [comment]
Thanks for including the full trace in the initial report

**Comment by egoldschmidt** [comment]
Happy to help! Thanks for the quick ACK

**Comment by egoldschmidt** [comment]
I believe that is the case, yes

**Comment by bdraco** [comment]
It looks like this can only happen if the `ClientResponse` is destroyed without being properly closed

**Comment by bdraco** [comment]
Its likely something isn't closing the WebSocket connections and they are only getting cleaned up in garbage collection.  Likely after cancellation in your case from the opening text.  If `loop.debug` is on you'll probably see an unclosed resource warning as well.

**Comment by bdraco** [comment]
It took me a while to figure out what why I couldn't replicate it in the tests. Our test harness was being a bit too helpful and cleaning up the unclosed WebSocket connections so we would never hit this path.

https://github.com/aio-libs/aiohttp/pull/9883

**Comment by bdraco** [comment]
3.11.2 published to PyPI with the fix in #9883
